src/GafferUI/Handle.cpp

```diff
@@ -556,18 +558,62 @@ Handle::AngularDrag::AngularDrag( bool processModifiers )
 }
 
 Handle::AngularDrag::AngularDrag( const Gadget *gadget, const Imath::V3f &origin, const Imath::V3f &axis0, const Imath::V3f axis1, const DragDropEvent &dragBeginEvent, bool processModifiers )
-	:	m_rotation( 0.0f ),
+	:	m_gadget( gadget ),
+		m_rotation( 0.0f ),
 		m_axis0( axis0 ),
 		m_axis1( axis1 ),
 		m_processModifiers( processModifiers ),
 		m_preciseMotionEnabled( false )
 {
 	// We need to negate this, or rotation is opposite to the mouse movement direction
 	V3f planeAxis0 = -axis0.cross( axis1 );
-	// Disable modifier processing as we'll do our own precision mode in angle space
-	m_drag = PlanarDrag( gadget, origin, planeAxis0, axis1, dragBeginEvent, false );
 
-	m_dragBeginRotation = closestRotation( m_drag.startPosition(), m_rotation );
+	WorldGadgetDragData dragData;
+	worldGadgetDragData( gadget, origin, planeAxis0, axis1, dragBeginEvent, dragData );
+
+	if( abs( dragData.worldPlane.normal.dot( dragData.worldLine.dir ) ) < g_planarToLinearDragThreshold )
+	{
+		// Do a LinearDrag in raster space instead of world space
+
+		float axis0Dot = dragData.worldAxis0.dot( dragData.worldLine.dir );
+		float axis1Dot = dragData.worldAxis1.dot( dragData.worldLine.dir );
+
+		bool useAxis0 = abs( axis0Dot ) < abs( axis1Dot );
+
+		// Looking down the normal of the rotation plane, positive values rotate
+		// counterclockwise. Find the gadget-space vector pointing in the direction
+		// of positive rotation to use for the start direction of the linear drag.
+		V3f parallelAxis = useAxis0 ? axis1 : planeAxis0;
```

Comment:
This naming feels confusing ... useAxis0 means it will use axis1, not axis0?

Reply:
Yes, "use" is too general here, is `int dragAxis = [0,1]` clearer? That's what I did in https://github.com/GafferHQ/gaffer/pull/5281/commits/b015786d886bd64a2e3299b445643ac3c05d4071